src/ds/d500/d500-auto-calibration-handler.cpp

```diff
@@ -0,0 +1,139 @@
+    }
+
+    d500_calibration_answer d500_auto_calibrated_handler::get_status() const
+    {
+        if (_dev)
+        {
+            auto cmd = _dev->build_command(ds::GET_CALIB_STATUS);
+            auto res = _dev->send_receive_raw_data(cmd);
+
+            // slicing 4 first bytes - opcode
+            res.erase(res.begin(), res.begin() + 4);
+            // checking size of received buffer
+            if (!check_buffer_size_from_get_calib_status(res))
+                throw std::runtime_error("GET_CALIB_STATUS returned struct with wrong size");
+
+            return *reinterpret_cast<d500_calibration_answer*>(res.data());
+        }
+        throw std::runtime_error("device has not been set");
+    }
+
+    std::vector<uint8_t> d500_auto_calibrated_handler::run_auto_calibration(d500_calibration_mode _mode)
+    {
+        if (_dev)
+        {
+            auto cmd = _dev->build_command(ds::SET_CALIB_MODE, static_cast<uint32_t>(_mode), 1 /*always*/);
+            return _dev->send_receive_raw_data(cmd);
+        }
+        throw std::runtime_error("device has not been set");
+    }
+
+    static std::vector<uint8_t> add_header_to_calib_config(const rs2_calibration_config& calib_config)
+    {
+        // calculate CRC
+        uint32_t computed_crc32 = rsutils::number::calc_crc32(reinterpret_cast<const uint8_t*>(&calib_config),
+            sizeof(rs2_calibration_config));
+
+        // prepare vector of data to be sent (header + sp)
+        rs2_calibration_config_with_header calib_config_with_header;
+        uint16_t version = ((uint16_t)0x01 << 8) | 0x01;  // major=0x01, minor=0x01 --> ver = major.minor
+        uint32_t calib_version = 0;  // ignoring this field, as requested by sw architect
+        calib_config_with_header.header = { version, static_cast<uint16_t>(ds::d500_calibration_table_id::calib_cfg_id),
+            sizeof(rs2_calibration_config), calib_version, computed_crc32 };
+        calib_config_with_header.payload = calib_config;
+        auto data_as_ptr = reinterpret_cast<uint8_t*>(&calib_config_with_header);
+        return std::vector<uint8_t>(data_as_ptr, data_as_ptr + sizeof(rs2_calibration_config_with_header));
+    }
+
+    void d500_auto_calibrated_handler::set_calibration_config(const rs2_calibration_config& calib_config)
+    {
+        if (_dev)
+        {
+            auto calib_config_with_header = add_header_to_calib_config(calib_config);
+
+            // prepare command
+            auto cmd = _dev->build_command(ds::SET_HKR_CONFIG_TABLE,
+                static_cast<int>(ds::d500_calib_location::d500_calib_flash_memory),
+                static_cast<int>(ds::d500_calibration_table_id::calib_cfg_id),
+                static_cast<int>(ds::d500_calib_type::d500_calib_dynamic), 0,
+                calib_config_with_header.data(), sizeof(rs2_calibration_config_with_header));
+
+            // send command 
+            auto res = _dev->send_receive_raw_data(cmd);
+        }
+        else
```

Comment:
Don't you preffer :
```
if ( !_dev )
    throw std::runtime_error("device has not been set");
```

then all the rest will not be indented and the throw will be close to the reason

Reply:
maybe less optimal if the compiler was simple - but more readable - so I agree - thanks